apps/sms/test/marionette/conversation_test.js

```diff
@@ -553,4 +566,114 @@ marionette('Conversation Panel Tests', function() {
+      assert.include(recipients, '+123', '+123 should be in the recipients');
+      assert.include(recipients, '+234', '+234 should be in the recipients');
+      assert.equal(newMessageView.messageText, '', 'MMS text should be empty');
+    });
+
+    test('Sms being written: message and recipients should be kept',
+    function() {
+      conversationView.typeMessage('message2');
+      var newMessageView = conversationView.includeSomeoneElse();
+      assert.equal(newMessageView.recipients, '+123',
+        'Recipient should be +123');
+      assert.equal(newMessageView.messageText, 'message2',
+        'message text should be kept');
+    });
+
+    test('Mms being written: all info must be kept', function() {
+      conversationView.typeMessage('message2');
+      // add a subject
+      conversationView.showSubject();
+      conversationView.typeSubject('subject1');
+      var newMessageView = conversationView.includeSomeoneElse();
+
+      // assert that we kept all the infos
+      assert.equal(newMessageView.recipients, '+123',
+        'Recipient should be +123');
+      assert.equal(newMessageView.messageText, 'message2',
+        'Message text should be kept');
+      assert.equal(newMessageView.subject, 'subject1',
+        'Subject should be kept');
+
+      // let's add a recipients and send the message.
+      // we will have a thread with several recipients
+      newMessageView.addNewRecipient('+234');
+      conversationView = newMessageView.send();
+      // write another MMS
+      conversationView.typeMessage('message3');
+      conversationView.showSubject();
+      conversationView.typeSubject('subject2');
+      // add a blob
+      conversationView.addAttachment().choose('Messages Activity Caller')
+        .pickImage();
+      messagesApp.switchTo();
+      // take screenshot of current attachment to compare it later
+      var beforeScreenshot = conversationView
+        .takeComposerAttachmentScreenshot(0);
+
+      // redirect to new message to add recipients
+      newMessageView = conversationView.includeSomeoneElse();
+
+      // assert the content
+      var recipients = newMessageView.recipients;
+      assert.lengthOf(recipients, 2, 'There should be 2 recipients');
+      assert.include(recipients, '+123', '+123 should be in the recipients');
+      assert.include(recipients, '+234', '+234 should be in the recipients');
+      assert.equal(newMessageView.messageText, 'message3',
+        'Message text should be kept');
+      assert.equal(newMessageView.subject, 'subject2',
+        'Subject should be kept');
+
+      // hide the keyboard: when it is displayed, the attachment is hidden,
+      // which prevents the screenshot to be taken
+      client.switchToFrame();
+      keyboardApp.switchTo();
+      keyboardApp.longPressSpaceBar(0.7);
+      client.switchToFrame();
+      // wait for the keyboard to be hidden
+      var systemInputMgmt = client.loader.getAppClass(
```

Comment:
is it ok to have a local var for that? As it is used only here, I thought I don't need to expose that.